### PR TITLE
support more CVTerms and annotations

### PR DESCRIPTION
**Pull request by exaexa**
Mainly a dependency of https://github.com/LCSB-BioCore/COBREXA.jl/pull/750 .

### CODE REVIEW

File: src/structs.jl [hunk 1]
```diff
@@ -479,4 +484,6 @@ Base.@kwdef struct Model
     volume_units::Maybe{String} = nothing
     notes::Maybe{String} = nothing
     annotation::Maybe{String} = nothing
+    sbo::Maybe{String} = nothing
+    cv_terms::Vector{CVTerm} = []
```
Comment (giordano) [inline src/structs.jl:488]: You may want to add these two fields to the tests in https://github.com/LCSB-BioCore/SBML.jl/blob/0eb0a446b62b5072e851d61fcab1e2e0c66b8124/test/writemodels.jl#L58

File: src/structs.jl [hunk 2]
```diff
@@ -208,6 +208,8 @@ Base.@kwdef struct Parameter
     units::Maybe{String} = nothing
     constant::Maybe{Bool} = nothing
     metaid::Maybe{String} = nothing
+    notes::Maybe{String} = nothing
+    annotation::Maybe{String} = nothing
```
Comment (giordano) [inline src/structs.jl:212]: I was going to suggest to add `Parameter` to the list of structs with annotations in the tests, but somehow it was already there: https://github.com/LCSB-BioCore/SBML.jl/blob/0eb0a446b62b5072e851d61fcab1e2e0c66b8124/test/common.jl#L25-L29 :sweat_smile:

Comment (giordano) [reply src/structs.jl:212]: We could actually use the same `==` definition (the one which checks for `annotations` and `notes`) for all structs, but I think it'll be slightly slower for structs without annotations.  Performance is probably not a huge concern here though.

File: src/structs.jl (hunk 1 above)
Comment (exaexa) [reply src/structs.jl:488]: I actually systematized it to whole model, that way it's much easier.

Comment (giordano) [reply src/structs.jl:488]: Uh, yeah, good idea.  I think I added each field manually there before having the smarter custom `==` method for the other structs.

**Review by giordano — APPROVED** [review]

**Pull request merged**